app/helpers/pages_helper.rb

```diff
@@ -9,6 +9,11 @@ def page_nav_item(text, path, strict = true)
     end
   end
 
+  def format_ak_ui_url(api_url)
+    return api_url unless Settings.ak_ui_url.present?
+    api_url.gsub(%r{.*rest\/v1}, Settings.ak_ui_url)
+  end
```

Comment:
What you suggest allows easier testing, but on the other hand there's not going to be a single place in the codebase where you call this method with a different `ak_domain` than the one in settings, and it does makes the method call from the views longer/uglier.